apps/src/lab2/levelEditors/levelData/EditToolboxBlocks.tsx

```diff
@@ -0,0 +1,155 @@
+import classNames from 'classnames';
+import {isEqual} from 'lodash';
+import React, {useCallback} from 'react';
+
+import {Button} from '@cdo/apps/componentLibrary/button';
+import Checkbox from '@cdo/apps/componentLibrary/checkbox/Checkbox';
+import {BodyTwoText} from '@cdo/apps/componentLibrary/typography';
+import {BlockTypes} from '@cdo/apps/music/blockly/blockTypes';
+import {
+  categoryTypeToLocalizedName,
+  dynamicCategories,
+} from '@cdo/apps/music/blockly/toolbox';
+import {defaultMaps} from '@cdo/apps/music/blockly/toolbox/definitions';
+import toolboxBlocks from '@cdo/apps/music/blockly/toolbox/toolboxBlocks';
+import {
+  Category,
+  CategoryBlocksMap,
+  ToolboxType,
+} from '@cdo/apps/music/blockly/toolbox/types';
+import {BlockMode} from '@cdo/apps/music/constants';
+import {getTypedKeys, ValueOf} from '@cdo/apps/types/utils';
+
+import MultiCategorySelect from './MultiCategorySelect';
+
+import styles from './edit-music-level-data.module.scss';
+
+interface EditToolboxBlocksProps {
+  blocksMap: CategoryBlocksMap;
+  onChange: (blocksMap: CategoryBlocksMap) => void;
+  blockMode: ValueOf<typeof BlockMode>;
+  toolboxType?: ToolboxType;
+}
+
+/**
+ * Editor for selecting allowed blocks and categories in the Music Lab toolbox.
+ */
+const EditToolboxBlocks: React.FC<EditToolboxBlocksProps> = ({
+  blocksMap,
+  onChange,
+  blockMode,
+  toolboxType,
+}) => {
+  const defaultBlocks = defaultMaps[blockMode];
+  const includedDynamicCategories =
+    toolboxType === 'flyout'
+      ? []
+      : dynamicCategories.filter(category => defaultBlocks[category]);
+
+  const toggleBlock = useCallback(
+    (category: Category, block: BlockTypes | string, checked: boolean) => {
+      const newBlocksMap = {...blocksMap};
+      if (!newBlocksMap[category]) {
+        newBlocksMap[category] = [];
+      }
+      if (checked) {
+        newBlocksMap[category].push(block);
+      } else {
+        newBlocksMap[category] = newBlocksMap[category].filter(
+          b => b !== block
+        );
+      }
+
+      if (newBlocksMap[category].length === 0) {
+        delete newBlocksMap[category];
+      }
+      onChange(newBlocksMap);
+    },
+    [blocksMap, onChange]
+  );
+
+  const toggleCategory = useCallback(
+    (category: Category, checked: boolean) => {
+      const newBlocksMap = {...blocksMap};
+      if (checked) {
+        newBlocksMap[category] = defaultBlocks[category] || [];
+      } else {
+        delete newBlocksMap[category];
+      }
+      onChange(newBlocksMap);
+    },
+    [blocksMap, onChange, defaultBlocks]
+  );
+
+  const multiSelectItems = getTypedKeys(defaultBlocks)
+    .filter(category => !dynamicCategories.includes(category))
+    .map(category => {
+      const categoryItems = (defaultBlocks[category] || []).map(block => {
```

Comment:
Nit: It looks like `block` is a string and one of our `BlockTypes`. Since it's not actually a block (or even a toolbox item representation of one), could we rename it to `blockType`? Seeing it assigned to `id` and `label` below might feel cleaner because we'd expect `blockType` to be a string.

Reply:
Good call, done